DOC-7888 update RBAC for collection admin

osfameron opened the pull request.

osfameron committed:
DOC-7888 update RBAC for collection admin

maria-robobug commented on package.json:
```diff
@@ -0,0 +1,5 @@
+{
+  "dependencies": {
+    "couchbase": "github:couchbase/couchnode#master"
```
Should this not point to the latest 3.1.X version(tag) instead of master? Or else I think we'll be pulling the latest dev version of the SDK which seems to be [3.2.0](https://github.com/couchbase/couchnode/blob/master/package.json#L58).

osfameron replied:
hmm, good point, with 3.2.0 in the release/3.2 branch...
(there wasn't previously a package.json in the repo, just thought it made sense to commit mine, and looks like it was worth it to iron this out, good spot!)

osfameron replied:
OK, so the code sample doesn't work with 3.1.3, I assume it'll be backported eventually, but may be simpler to keep master for now...?

Would be good to have our example automation verify which versions work correctly for different snippets perhaps?

maria-robobug commented on modules/howtos/examples/collection-manager.js:
```diff
@@ -3,66 +3,92 @@
+    await collectionMgr.dropCollection("example-collection", "example-scope");
+  }
+  catch (e) {
+    // if (e instanceof couchbase.CollectionNotFoundError) {
```
Should this be commented out?

osfameron replied:
see below

maria-robobug commented on modules/howtos/examples/collection-manager.js:
```diff
@@ -3,66 +3,92 @@
+  catch (e) {
+    // if (e instanceof couchbase.CollectionNotFoundError) {
+      console.log("The collection does not exist");
+    // }
+    // else if (e instanceof couchbase.ScopeNotFoundError) {
+    //   console.log("The scope does not exist");
+    // }
```
And these lines too

osfameron replied:
Yeah, these were commented previously:

```
❯ git blame modules/howtos/examples/collection-manager.js | grep '// else if '
0aeba097 (Hakim Cassimally 2021-03-19 17:04:06 +0000 36)       // else if (e instanceof couchbase.ScopeNotFoundError) {
0aeba097 (Hakim Cassimally 2021-03-19 17:04:06 +0000 50)       // else if (e instanceof couchbase.ScopeNotFoundError) {
❯ git show --no-patch 0aeba097
commit 0aeba09726bd706b74eeb3caab417a995fd24d27 (origin/doc-7968-collection-management)
Author: Hakim Cassimally <hakim.cassimally@couchbase.com>
Date:   Fri Mar 19 17:04:06 2021 +0000

    DOC-7968 document Collection Management

    NB: some of the error checking is commented out, pending JSCBC-858
```

I have a watch on https://issues.couchbase.com/browse/JSCBC-858
so hoping to update by 3.2.

maria-robobug replied:
ah gotcha, maybe add a comment like: `TODO: uncomment code once JSCBC-858 is resolved` or something along those lines.

osfameron replied:
ok, new commit using Asciidoc's tag filtering, how's that?

maria-robobug replied:
Love this approach! 👍

maria-robobug reviewed (approved):
👍  LGTM, just some minor comments.

osfameron committed:
DOC-7888 prune JSCBC-858 comments from code listings

Using Asciidocs nested tag filtering:
  https://docs.asciidoctor.org/asciidoc/latest/directives/include-tagged-regions/#tag-filtering
